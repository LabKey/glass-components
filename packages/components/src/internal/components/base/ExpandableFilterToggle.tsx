import React, { PureComponent, ReactNode } from 'react';
import classNames from 'classnames';

interface Props {
    filterExpanded: boolean;
    hasFilter: boolean;
    toggleFilterPanel: () => void;
    resetFilter: () => void;
    panelCls?: string;
}

export class ExpandableFilterToggle extends PureComponent<Props> {
    static defaultProps = {
        panelCls: 'show-hide-filter-toggle',
    };

    render(): ReactNode {
        const { filterExpanded, hasFilter, toggleFilterPanel, resetFilter, panelCls } = this.props;

        return (
            <>
                <div onClick={toggleFilterPanel} className={panelCls}>
                    {filterExpanded ? 'Hide filters ' : 'Show filters '}
                    <i
                        className={classNames('fa', {
                            'fa-chevron-down': filterExpanded,
                            'fa-chevron-right': !filterExpanded,
                        })}
                    />
                </div>
                {hasFilter && (
                    <span className="filter-clear-all" onClick={resetFilter}>
                        Clear All
                    </span>
                )}
            </>
        );
    }
}
