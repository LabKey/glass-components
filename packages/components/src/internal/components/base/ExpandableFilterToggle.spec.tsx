import React from 'react';
import renderer from 'react-test-renderer';

import { ExpandableFilterToggle } from './ExpandableFilterToggle';

describe('<ExpandableFilterToggle/>', () => {
    test('default properties without filters', () => {
        const component = (
            <ExpandableFilterToggle
                filterExpanded={false}
                hasFilter={false}
                toggleFilterPanel={jest.fn()}
                resetFilter={jest.fn()}
            />
        );
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('custom properties with filters', () => {
        const component = (
            <ExpandableFilterToggle
                filterExpanded={true}
                hasFilter={true}
                panelCls={'test-cls'}
                toggleFilterPanel={jest.fn()}
                resetFilter={jest.fn()}
            />
        );
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
