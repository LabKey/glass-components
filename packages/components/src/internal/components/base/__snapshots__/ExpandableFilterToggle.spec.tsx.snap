// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpandableFilterToggle/> custom properties with filters 1`] = `
Array [
  <div
    className="test-cls"
    onClick={[MockFunction]}
  >
    Hide filters 
    <i
      className="fa fa-chevron-down"
    />
  </div>,
  <span
    className="filter-clear-all"
    onClick={[MockFunction]}
  >
    Clear All
  </span>,
]
`;

exports[`<ExpandableFilterToggle/> default properties without filters 1`] = `
<div
  className="show-hide-filter-toggle"
  onClick={[MockFunction]}
>
  Show filters 
  <i
    className="fa fa-chevron-right"
  />
</div>
`;
